### ISSUE TITLE
Job creation fails w k8s 1.5beta1

### ISSUE THREAD

**Issue by sjezewski** [pachyderm/pachyderm#1193]
From the pachd logs we see:

```
2016-12-12T17:53:55Z ERROR error running pipeline filter: Job.batch "4ab2d81bf69bd012096aa7d95cc57467" is invalid: spec.template.spec.initContainers[0].env[1].valueFrom.fieldRef.apiVersion: Required value; retrying in 256.727409ms
2016-12-12T17:53:55Z ERROR error updating pipeline state: rpc error: code = 1 desc = context canceled
```

This is using:

```
$kubectl version
Client Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.0-beta.1", GitCommit:"8c6525e891be1c44cbdd6fcf53097a3adb11d68c", GitTreeState:"clean", BuildDate:"2016-11-18T22:06:00Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5+", GitVersion:"v1.5.0-beta.1", GitCommit:"8c6525e891be1c44cbdd6fcf53097a3adb11d68c", GitTreeState:"clean", BuildDate:"2016-11-18T22:01:26Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}```
$pachctl version
COMPONENT           VERSION             
pachctl             1.3.0               
pachd               1.3.0
```

This user needs 1.5 to be able to run k8s on azure, but it seems that we're not creating jobs properly for k8s 1.5

**Comment by abourget** [comment]
Yeah Pachyderm is just not useable on 1.5, because of this issue and [this issue](https://github.com/pachyderm/pachyderm/issues/1210)

**Comment by jdoliner** [comment]
Fixed by #1223.
